### PR TITLE
prep for merge into master

### PR DESCRIPTION
**Pull request by jakemac53**
- Updates travis config to run nnbd tests/analysis 
- Adds dependency overrides so tests can be ran in null safety mode
- Updates package version to expected pre-release version
- Updates sdk constraints to restrict to dev sdks with the allow list

### CODE REVIEW

**Comment by jakemac53** [comment]
~Note that the tests will be failing until https://github.com/dart-lang/test/pull/1283 is merged, so we probably want to hold off until then~

This is fixed now

**Review by lrhn — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Changelog
 
-## 1.1.0
+## 1.1.0-nnbd
```
Comment (lrhn) [inline CHANGELOG.md]: Maybe use `nullsafety` instead of `nnbd`. 

Insert "Who's NNBD, I'm Null Safety" meme.

Comment (jakemac53) [reply CHANGELOG.md]: I had originally tried null_safety but underscores aren't allowed, nullsafety one word is also probably fine though. cc @natebosch @kevmoo  wdyt?

Nothing is published yet so I can still change the other packages, we should try to be consistent tho.

Comment (natebosch) [reply CHANGELOG.md]: I have a slight preference for `-nullsafety`

Comment (jakemac53) [reply CHANGELOG.md]: talked with nate offline and we agree `-nullsafety` is better

**Pull request merged**